fix(run-name): update class methods for WorkflowHandler

evanlindsey opened the pull request:
When providing a `run-name`, the following error occurs: `Failed to get workflow url: Cannot read properties of undefined (reading '0')`. The root cause is that `check_runs` exists on the `result.data` object and not directly on the `result` variable (in `findWorklowRunIdFromRunName`). Initially, this PR was to fix the error condition and return value of this method. However, after further debugging, it was found that the returned `job id` was being incorrectly applied to the `octokit.rest.actions.getWorkflowRun` call in the `getWorkflowRunStatus` method (which is expecting a `run id`).

Additional updates were needed to return the full `WorkflowRunResult` of the `check_run` and have it correctly filtered for runs invoked after `this.triggerDate` (logic borrowed from `findWorkflowRunIdFromFirstRunOfSameWorkflowId`). Overall, conditions and return values for 3 class methods in `WorkflowHandler` were updated and 1 unused method was removed (`getWorkflowRunArtifacts`). The functionality of providing a `run-name` in the action config should now work as expected.

aurelien-baudet commented on src/workflow-handler.ts:
```diff
@@ -78,61 +78,37 @@ export class WorkflowHandler {
   }
 
-
-  async getWorkflowRunArtifacts(): Promise<WorkflowRunResult> {
```
Please do not remove code that you should not change with this PR.
You PR is to fix issue with runName, not cleaning code. Maybe this code is not used yet but could be later

aurelien-baudet reviewed (changes requested):
Please add some tests to ensure that your changes are ok

LudovicTOURMAN commented on src/workflow-handler.ts:
```diff
@@ -171,7 +147,7 @@ export class WorkflowHandler {
-  private async findWorklowRunIdFromRunName(runName: string): Promise<number> {
+  private async findWorklowRunFromRunName(runName: string): Promise<WorkflowRunResult> {
     const result = await this.octokit.rest.checks.listForRef({
       check_name: runName,
```
I tested your branch @evanlindsey and noticed it was still not working for me.

Having a deeper look, I figure out that current function `octokit.rest.checks.listForRef`, which is based on [`repos/OWNER/REPO/commits/REF/check-runs`](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-runs-for-a-git-reference), is not able to retrieve anything.

`check_name` is filled by Github with the [jobs.<job_id>](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_id) (or [jobs.<job_id>.name](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idname)) and we are filtering it with the [run-name](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name) which never will match :grimacing:

I'm really sad since Github is not puttin any effort to make this run id easily retrievable :disappointed:

evanlindsey replied:
@LudovicTOURMAN you're definitely right that `job name != run name`. In fact, I have this working on a couple of pipelines but am providing a unique `job name` from within the target `run`. I was just leveraging the existing function but realize this does not align with the readme and would mislead people. That said, I started to refactor for `run name` and noticed you opened a PR of your own. I'm fine to close this one and go with yours. However, I do have some observations and will leave comments. Thanks for the feedback!

evanlindsey commented:
> Thank you for your contribution.
> 
> Please add some tests to ensure that your changes are ok and create a non-regression test. 🙏

@aurelien-baudet thank you for the review! I replaced the removed function, simplified my changes, and added a test for `run name` in the `build-test` workflow. However, a newer PR has been opened to resolve the issue. I will close this one in favor of the other: https://github.com/aurelien-baudet/workflow-dispatch/pull/17
